gazebo_plugins/src/gazebo_ros_depth_camera.cpp

```diff
@@ -156,6 +171,24 @@ void GazeboRosDepthCamera::PointCloudDisconnect()
     this->parentSensor->SetActive(false);
 }
 
+////////////////////////////////////////////////////////////////////////////////
+// Increment count
+void GazeboRosDepthCamera::NormalsConnect()
+{
+  this->normals_connect_count_++;
+  (*this->image_connect_count_)++;
+  this->parentSensor->SetActive(true);
+}
+////////////////////////////////////////////////////////////////////////////////
+// Decrement count
+void GazeboRosDepthCamera::NormalsDisconnect()
+{
+  this->normals_connect_count_--;
+  (*this->image_connect_count_)--;
+  if (this->normals_connect_count_ <= 0)
+    this->parentSensor->SetActive(false);
```

Comment:
hmm if there are still subscribers to the point cloud topic (or other topics advertised by the depth camera), this line will disable the sensor causing no msgs to be published. I think we should disable the sensor only when there are no subscribers to all the topics